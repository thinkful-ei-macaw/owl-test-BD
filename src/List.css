.chatbar-avatar {
  border: solid 2px;
  border-radius: 10rem;
  height: 50px;
  width: 50px;
  float: left;
}
li.User {
  display: block;
}

h2 {
  padding-left: 10px;
}

p {
  display: flex;
  flex-direction: row;
}

ul{
  list-style-type: disc;
}

li.Empty {
  color: green;
  content: "\2022";
}

