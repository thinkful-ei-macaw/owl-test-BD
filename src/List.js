import React from 'react';
import './List.css';


function List(props) {

let status = "";
if (props.onStage === true) {
  status = "On Stage";
} else if (props.onStage === false && props.inSession === true) {
  status = "In Session"
} else {
  status = "Left Session"
}

return (
    <li className="User" id={props.id}>
      <img src={props.avatar} alt="" className='chatbar-avatar'></img>
      <h2>{props.name}</h2>
      <p className="Status"><ul><li className="Empty"></li></ul>{status}</p>
    </li>
 )
}

export default List;