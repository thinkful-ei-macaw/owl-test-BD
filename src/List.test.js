import React from 'react';
import ReactDOM from 'react-dom'
import List from './List'
import renderer from 'react-test-renderer'

describe('<List />', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');

      // Pass required props in the usual way
      ReactDOM.render(
          <List
              name="Joe"
              avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
              inSession={true}
              onStage={true}
          />,
          div
      );

      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders this UI as expected', () => {
    // Render the component, as JSON
    const tree = renderer.create(<List
      name="Joe"
      avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      inSession={true}
      onStage={true}
  />).toJSON();
    // Check whether it matches the previous snapshot
    // Stored in __snapshots__/App.test.js.snap
    expect(tree).toMatchSnapshot(); 
});
});
