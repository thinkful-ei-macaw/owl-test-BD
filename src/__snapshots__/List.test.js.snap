// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<li
  className="User"
>
  <img
    alt=""
    className="chatbar-avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <h2>
    Joe
  </h2>
  <p
    className="Status"
  >
    <ul>
      <li
        className="Empty"
      />
    </ul>
    On Stage
  </p>
</li>
`;
